### PR TITLE
feat(core): desktop multiple server support

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8979** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8979?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8979?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/modules/telemetry/services/telemetry.ts [hunk 1]
```diff
@@ -2,27 +2,39 @@ import { mixpanel } from '@affine/track';
...
+      )
+    );
+    this.disposableFns.push(
+      defaultServerService.server.eventBus.on(AccountChanged, () =>
+        this.onAccountLoggedOut()
+      )
+    );
   }
 
   onApplicationStart() {
```
Comment (forehalo) [inline packages/frontend/core/src/modules/telemetry/services/telemetry.ts:54]: <i class='graphite__hidden'>[Re: line +59]<br/><br/></i>

reset when switching to new user<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8979?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by forehalo — APPROVED** [review]

File: packages/frontend/apps/electron/src/main/protocol.ts [hunk 2]
```diff
@@ -83,31 +83,56 @@ export function registerProtocol() {
...
             }
-            return newCookie;
-          });
-        }
-      }
+          }
 
-      callback({ responseHeaders });
+          responseHeaders['Access-Control-Allow-Origin'] = ['*'];
+          responseHeaders['Access-Control-Allow-Headers'] = ['*'];
```
Comment (pengx17) [inline packages/frontend/apps/electron/src/main/protocol.ts:120]: do we need this?

Comment (EYHN) [reply packages/frontend/apps/electron/src/main/protocol.ts:120]: This is to bypass CORS

Comment (pengx17) [reply packages/frontend/apps/electron/src/main/protocol.ts:120]: Need to consider the response header have lowercase header names
```
delete responseHeaders['access-control-allow-origin'];
delete responseHeaders['access-control-allow-headers'];
```

File: packages/frontend/apps/electron/src/main/protocol.ts [hunk 3]
```diff
@@ -148,22 +170,33 @@ export function registerProtocol() {
...
+        url.protocol === 'http:' ||
+        url.protocol === 'https:' ||
+        url.protocol === 'ws:' ||
+        url.protocol === 'wss:'
+      ) {
+        const cookies = await session.defaultSession.cookies.get({
+          url: details.url,
+        });
+
+        const cookieString = cookies
```
Comment (pengx17) [inline packages/frontend/apps/electron/src/main/protocol.ts:179]: Need to consider cases that the request headers already have `Cookie`/`cookie` fields.

**Review by pengx17 — APPROVED** [review]
`CLOUD_BASE_URL` in electron should be removed.

**Comment by Felitendo** [comment]
Thank you so much for finally adding this! <3

**Pull request merged**